### PR TITLE
Implement search page

### PR DESCRIPTION
**Pull request by abukurov**

### CODE REVIEW

File: app/components/pagination-pager/template.hbs [hunk 1]
```diff
@@ -1,10 +1,10 @@
 {{#if (gt totalPages 1)}}
   {{#if (gt currentPage 1)}}
-    {{#pagination-item page=1 pageId="first" onClick=(action setCurrentPage)}}
+    {{#pagination-item page=1 pageId="first" onClick=(action "setCurrentPage")}}
```
Comment (maxmalov) [inline app/components/pagination-pager/template.hbs]: Shouldn't it work without quotes?

File: app/components/provider-status/component.js [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import Ember from 'ember';
 
 export default Ember.Component.extend({
+  tagName: 'i',
```
Comment (maxmalov) [inline app/components/provider-status/component.js]: How about http://fontawesome.io/icon/circle/

File: app/provider/route.js [hunk 3]
```diff
@@ -1,4 +1,19 @@
...
+} = Ember;
+
 export default Ember.Route.extend({
+  provider: service(),
+
+  model(params) {
+    return this.get('provider').get(params.providerId);
+  },
+
+  redirect(model, transition) {
```
Comment (maxmalov) [inline app/provider/route.js]: why is this redirect needed?

File: app/search/controller.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import Ember from 'ember';
+
+const { computed } = Ember;
+
+export default Ember.Controller.extend({
+  queryParams: ['page', 'query'],
+
+  totalPages: computed.reads('model.meta.totalPages'),
```
Comment (maxmalov) [inline app/search/controller.js]: This property can be omitted. Spawning computed props for everything can be a performance penalty

File: app/search/controller.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+  currentPage: computed('page', function () {
```
Comment (maxmalov) [inline app/search/controller.js]: What do you think if we stick to prototype extensions?

File: app/search/route.js [hunk 6]
```diff
@@ -1,8 +1,29 @@
...
+  queryParams: {
+    page: {
+      refreshModel: true
+    }
+  },
+
+  provider: service(),
+
+  model(query) {
+    return this.get('provider').list(query);
```
Comment (maxmalov) [inline app/search/route.js]: I propose to rename `list` to `getPage`

File: app/search/template.hbs [hunk 7]
```diff
@@ -1,6 +1,14 @@
 <div class="search">
   {{#content-title}}{{model.title}}{{/content-title}}
-  {{#each model.result as |item|}}
+  {{#search-results
+    currentPage=currentPage
+    totalPages=totalPages
+    items=model.items
+    setCurrentPage=(action "setCurrentPage")
```
Comment (maxmalov) [inline app/search/template.hbs]: `onPageChange=(action ...)`

File: app/components/provider-status/component.js (hunk 2 above)
Comment (maxmalov) [inline app/components/provider-status/component.js]: P.S. This is why unrelated parts of code shouldn't be touched)))

File: app/components/pagination-pager/template.hbs (hunk 1 above)
Comment (abukurov) [inline app/components/pagination-pager/template.hbs]: it doesn't work when action isn't specified

File: app/provider/route.js (hunk 3 above)
Comment (abukurov) [inline app/provider/route.js]: To display the default route

**Pull request merged**